railties/test/application/rake/dev_test.rb

```diff
@@ -17,29 +17,29 @@ def teardown
 
       test "dev:cache creates file and outputs message" do
         Dir.chdir(app_path) do
-          output = rails("dev:cache")
+          output = `bin/rake dev:cache`
           assert File.exist?("tmp/caching-dev.txt")
           assert_match(/Development mode is now being cached/, output)
         end
       end
 
       test "dev:cache deletes file and outputs message" do
         Dir.chdir(app_path) do
-          rails "dev:cache" # Create caching file.
-          output = rails("dev:cache") # Delete caching file.
+          `bin/rake dev:cache` # Create caching file.
+          output = `bin/rake dev:cache` # Delete caching file.
           assert_not File.exist?("tmp/caching-dev.txt")
           assert_match(/Development mode is no longer being cached/, output)
         end
       end
 
       test "dev:cache touches tmp/restart.txt" do
         Dir.chdir(app_path) do
-          rails "dev:cache"
+          `bin/rake dev:cache`
           assert File.exist?("tmp/restart.txt")
 
           prev_mtime = File.mtime("tmp/restart.txt")
           sleep(1)
-          rails "dev:cache"
+          `bin/rake dev:cache`
```

Comment:
Would be good to assert deprecation warning in each test of this file

Reply:
I think we can use

```ruby
stderr = capture(:stderr) do
  # rake ...
end

assert_match(/DEPRECATION WARNING: .../, stderr)
```
I had similar case in https://github.com/rails/rails/commit/a6ebafc5ac71dd11a548ac884dc1637a1a5db926

Reply:
Oh thanks for complementing my change to `NotesCommand` ✨ 😄 . I'll capture stdout. Thanks for the tip